hcpannot/analysis/lmm.py

```diff
@@ -0,0 +1,222 @@
+    '''
+    Postprocess the result of the linear mixed-effects model. The function sets the variance explained by researchers to 0
+    and the variance explained by subjects to 1 for the first data point of the hV4_outer and VO_outer contours.
+    
+    The function also calculates the residual variance (variance not explained by researchers or subjects).
+    '''
+    
+    for contour in ['hV4_outer', 'VO_outer']: 
+        # the starting point of these two contours is the same for all researchers if fsaverage data is used
+        # therefore, the variance explained by researchers is conceptually 0 at data point 1
+        
+        for hemi in ['lh', 'rh']:
+            
+            if not native_aligned: # if fsaverage data is used
+                
+                # check if the result for data point 1 of the contour exists
+                exists = ((result['contour'] == contour) & (result['loc_on_contour'] == 1) & (result['hemi'] == hemi)).any()
+
+                if exists: 
+                    # if the result exists, set the variance explained by researchers to 0
+                    result.loc[(result['contour']== contour) & (result['loc_on_contour']==1) & 
+                                   (result['hemi']==hemi), 'varex_rater'] = 0
+                    # set the variance explained by subjects to 1
+                    result.loc[(result['contour']== contour) & (result['loc_on_contour']==1) & 
+                                   (result['hemi']==hemi), 'varex_sbj'] = 1
+
+                else:  
+                    # if the result does not exist (e.g. model failed to converge), add a new row to the dataframe
+                    # where the variance explained by researchers is 0 and the variance explained by subjects is 1
+                    new_row = pd.DataFrame([[0,1,hemi,contour,1]], columns=['varex_rater', 'varex_sbj', 
+                                                                                'hemi', 'contour', 'loc_on_contour'])
+                    # append the new row to the dataframe
+                    result=pd.concat([result,new_row], ignore_index=True)
+    
+    # sort the dataframe by contour, hemisphere, and location on the contour
+    result = result.sort_values(by=['contour', 'hemi', 'loc_on_contour'], ignore_index=True)
+    
+    # calculate the residual variance (variance not explained by researchers or subjects)
+    result['residual'] = 1 - result['varex_rater'] - result['varex_sbj']
+    
+    return result
+
+def get_contour_data (result, contour): # get the data for a specific contour
+    
+    '''
+    Get the data for a specific contour. The function returns a dictionary that contains the variance explained by researchers,
+    the variance explained by subjects, and the residual variance for the left and right hemisphere.
+    '''
+
+    # create a dictionary to store the data
+    contour_data = defaultdict(lambda: defaultdict(list))
```

Comment:
I think this can just be `defaultdict(dict)` since you don't actually use `contour_data[hemi][tag].append(...)` anywhere.